tests/unittests/tests-libc/tests-libc.c

```diff
@@ -30,6 +30,38 @@ static void test_libc_strscpy(void)
     TEST_ASSERT_EQUAL_INT(strscpy(buffer, "empty", 0), -E2BIG);
 }
 
+static void test_libc_swprintf(void)
+{
+    string_writer_t sw;
+    char buffer[32];
+    int res;
+
+    string_writer_init(&sw, buffer, sizeof(buffer));
+
+    /* check that string can be written completely */
+    res = swprintf(&sw, "Hello World!");
+    TEST_ASSERT_EQUAL_INT(res, 12);
+    TEST_ASSERT_EQUAL_INT(string_writer_len(&sw), 12);
+    TEST_ASSERT_EQUAL_INT(strcmp("Hello World!", string_writer_str(&sw)), 0);
+
+    /* check that we can add to the string */
+    res = swprintf(&sw, "0123456789");
+    TEST_ASSERT_EQUAL_INT(res, 10);
+
+    /* check string truncation */
+    res = swprintf(&sw, "01234567891");
+    TEST_ASSERT_EQUAL_INT(res, -E2BIG);
+
+    /* check writing to full buffer */
+    res = swprintf(&sw, "###");
+    TEST_ASSERT_EQUAL_INT(res, -E2BIG);
```

Comment:
```suggestion
    /* The string does not fit completely into the buffer, so it gets truncated */
    res = swprintf(&sw, "01234567891");
    TEST_ASSERT_EQUAL_INT(res, -E2BIG);
    TEST_ASSERT_EQUAL_INT(strcmp("Hello World!0123456789012345678", string_writer_str(&sw)), 0);

    /* You can't write to a full buffer */
    res = swprintf(&sw, "###");
    TEST_ASSERT_EQUAL_INT(res, -E2BIG);
```

(maybe change the 012..89 with words to make it easier to comprehend?)

Reply:
the numbers make it easier to see how many bytes were actually written